src/mem/pagemap.h

```diff
@@ -10,6 +10,36 @@ namespace snmalloc
   static constexpr size_t PAGEMAP_NODE_BITS = 16;
   static constexpr size_t PAGEMAP_NODE_SIZE = 1ULL << PAGEMAP_NODE_BITS;
 
+  /**
+   * Structure describing the configuration of a pagemap.  When querying a
+   * pagemap from a different instantiation of snmalloc, the pagemap is exposed
+   * as a `void*`.  This structure allows the caller to check whether the
+   * pagemap is of the format that they expect.
+   */
+  struct PagemapConfig
+  {
```

Comment:
Should address bits be part of this?  We could end up exposing across a 32/64 bit boundary?  Many things would go wrong.  But perhaps this would catch it early.

Reply:
Good idea, added